src/plone/app/multilingual/browser/selector.py

```diff
@@ -72,27 +76,55 @@ class LanguageSelectorViewlet(LanguageSelector):
 
     def languages(self):
         languages_info = super(LanguageSelectorViewlet, self).languages()
+        registry = getUtility(IRegistry)
+        always_show_selector = registry['plone.always_show_selector']
         results = []
+        translations = []
         translation_group = queryAdapter(self.context, ITG)
+
         if translation_group is None:
             translation_group = NOTG
+
+        # for some reason there is no ITranslationManager provided by
```

Comment:
I still don't understand. In you example above the adapter is properly found and gives correct languages.